lib/gooddata/models/project.rb

```diff
@@ -1757,7 +1758,10 @@ def remove_users(list)
     end
 
     def check_groups(specified_groups, options = {})
-      groups = user_groups.map(&:name)
+      if @user_groups_cache.nil?
```

Comment:
Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.